napalm_eos/eos.py

```diff
@@ -1201,3 +1201,143 @@ def traceroute(self, destination, source='', ttl=0, timeout=0):
+        if neighbor_address:
+            peer_info = self._parse_per_peer_bgp_detail(
+                raw_output[0]['output'])
+
+            # Appending peer_info to final data structure
+            if int(peer_info[0]['remote_as']) not in bgp_detail_info:
+                bgp_detail_info[int(peer_info[0]['remote_as'])] = []
+
+            (bgp_detail_info[int(peer_info[0]['remote_as'])].
+                append(peer_info[0]))
+
+        else:
+            # Using preset template to extract peer info
+            v4_peer_info = self._parse_per_peer_bgp_detail(
+                raw_output[0]['output'])
+
+            v6_peer_info = self._parse_per_peer_bgp_detail(
+                raw_output[1]['output'])
+
+            for peer_info in v4_peer_info:
+                # Appending v4 peer_info to final data structure
+                if int(peer_info['remote_as']) not in bgp_detail_info:
+                    bgp_detail_info[int(peer_info['remote_as'])] = []
+
+                bgp_detail_info[int(peer_info['remote_as'])].append(peer_info)
+
+            for peer_info in v6_peer_info:
+                # Appending v6 peer_info to final data structure
+                if int(peer_info['remote_as']) not in bgp_detail_info:
+                    bgp_detail_info[int(peer_info['remote_as'])] = []
+
+                bgp_detail_info[int(peer_info['remote_as'])].append(peer_info)
+
+        return bgp_detail_info
+
+    def _parse_per_peer_bgp_detail(self, peer_output):
+        """This function parses the raw data per peer and returns a
+        json structure per peer.
+        """
+
+        int_fields = ['accepted_prefix_count', 'local_as', 'remote_as',
+                      'local_port', 'remote_port', 'local_port',
+                      'input_messages', 'output_messages', 'input_updates',
+                      'output_updates', 'messages_queued_out', 'holdtime',
+                      'configured_holdtime', 'keepalive',
+                      'configured_keepalive', 'advertise_prefix_count',
+                      'received_prefix_count']
+
+        peer_details = []
+
+        # Using preset template to extract peer info
+        peer_info = (
+            napalm_base.helpers.textfsm_extractor(
+                self, 'bgp_detail', peer_output))
+
+        for item in peer_info:
+            # Issuing command to get number of accepted prefixes for peer
+            try:
+                peer_ver = IPAddress(item['remote_address']).version
+            except Exception:
+                continue
+
+            if peer_ver == 4:
+                acc_prefix_command = (
+                    'show ip bgp summary | include %s' %
+                    item['remote_address'])
+            elif peer_ver == 6:
+                acc_prefix_command = (
+                    'show ipv6 bgp summary | include %s' %
+                    item['remote_address'])
+
+            accepted_prefix_output = (
+                self.device.run_commands([acc_prefix_command],
```

Comment:
I have a better idea: instead of issuing one command and instructing the device to parse the output, why not retrieving again the whole output of ```sh ip bgp sum``` (+```sh ipv6 bgp sum```) and parsing (probably you'll need to define a second template). Will result a hash with the mapping between the neighbor IP Address and ```PfxRcd``` and ```PfxAcc```. Then you'll just need to extract this data from the dictionary.